Api/V8/Service/ModuleService.php

```diff
@@ -113,39 +116,92 @@ public function getRecords(GetModulesParams $params, Request $request)
         $realRowCount = $this->beanManager->countRecords($module, $where);
         $limit = $size === BeanManager::DEFAULT_ALL_RECORDS ? BeanManager::DEFAULT_LIMIT : $size;
         $deleted = $params->getDeleted();
+        $data = [];
 
         if (empty($fields)) {
             $fields = $this->beanManager->getDefaultFields($bean);
         }
 
-        $beanListResponse = $this->beanManager->getList($module)
-            ->orderBy($orderBy)
-            ->where($where)
-            ->offset($offset)
-            ->limit($limit)
-            ->max($size)
-            ->deleted($deleted)
-            ->fields($this->beanManager->filterAcceptanceFields($bean, $fields))
-            ->fetch();
-
-
-        $beanArray = [];
-        foreach ($beanListResponse->getBeans() as $bean) {
-            $bean = $this->beanManager->getBeanSafe(
-                $params->getModuleName(),
-                $bean->id
-            );
-            $beanArray[] = $bean;
+
+        // Detect if bean has email field
+        if ((property_exists($bean, 'email1') && strpos($where, 'email1')) || (property_exists($bean,
+                    'email2') && strpos($where, 'email2'))) {
+            $selectedFields = strtolower($module) . '.' . implode(',' . substr(strtolower($module), 0, 1) . '.', $fields);
+            $selectedModule = strtolower($module);
+
+            //Selects Module or COUNT(*) and will add one to the query.
+            $idSelect = "SELECT {$selectedModule}.id ";
+            $countSelect = 'SELECT COUNT(*) AS cnt ';
+
+            //Email where clause
+            $fromQuery = "FROM email_addresses join email_addr_bean_rel on email_addresses.id = email_addr_bean_rel.email_address_id join {$selectedModule} on {$selectedModule}.id = email_addr_bean_rel.bean_id ";
+            $modifiedWhere = str_replace("accounts.email1", "email_addresses.email_address", $where);
+            $where = "$modifiedWhere";
+
+            //Sets and adds deleted to the query
```

Comment:
Minor coding standard issues with a few comments. They should start with a capital letter + space after the ```//```.
  
```suggestion
            // Sets and adds deleted to the query
```